### PR TITLE
Todo list react App

### PR DESCRIPTION
**Pull request by brahimdidi**
In this app, I used React class components and then converted it to react functional Components using hooks.

## Requierements

Use JavaScript events.
Use React life cycle methods.
Understand the mechanism of lifting the state up.

### CODE REVIEW

**Comment by brahimdidi** [comment]
ES lint to be fixed later

**Comment by preciousakams** [comment]
Hi @brahimdidi   😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💪🏻
- Your code is clean.
- Your code is functioning as expected✔️
- GitFlow best practices have been followed ✔️
- You have a descriptive PR ✔️
👍 Good Job

## Required Changes ♻️
- You have linters error, kindly fix this

- Check the comments under and make all required changes

## Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Cheers and Happy coding!👏👏👏 💻
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @kamzzy when you comment so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

File: src/functionComp/TodoItem.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  
+
+    const completedStyle = {
+        fontStyle: "italic",
+        color: "#595959",
+        opacity: 0.4,
+        textDecoration: "line-through",
+      }
+      const { completed, id, title } = props.todo;
```
Comment (kamzzy) [inline src/functionComp/TodoItem.js:24]: You have a prop validation error, kindly add propTypes to solve this

Comment (preciousakams) [inline src/functionComp/TodoItem.js:24]: You have prop validation error, kindly add propTypes to solve this

File: src/functionComp/TodoItem.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      return (
+        <li className={styles.item}>
+        <div onDoubleClick={handleEditing} style={viewMode}>
+          <input
+            type="checkbox"
+            className={styles.checkbox}
+            checked={completed}
+            onChange={() => props.handleChangeProps(id)}
+          />
+          <button onClick={() => props.deleteTodoProps(id)}>
```
Comment (preciousakams) [inline src/functionComp/TodoItem.js:51]: add propTypes as well

File: src/functionComp/InputTodo.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const onChange = e => {
+    setInputText({
+      ...inputText,
+      [e.target.name]: e.target.value,
+    })
+  }
+  const handleSubmit = e => {
+    e.preventDefault();
+    if (inputText.title.trim()) {
+        props.addTodoProps(inputText.title)
```
Comment (preciousakams) [inline src/functionComp/InputTodo.js:16]: add propType here as well

File: src/functionComp/InputTodo.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        setInputText({
+          title: "",
+        }) 
+      } else {
+        alert("Please write item")
```
Comment (preciousakams) [inline src/functionComp/InputTodo.js:21]: using alerts is against javascript best practices, kindly use  another element, for instance, a modal

File: src/functionComp/TodosList.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import TodoItem from "./TodoItem";
+const TodosList = (props) => {
+  
+    return (
+      <ul>
+        {props.todos.map(todo => (
```
Comment (preciousakams) [inline src/functionComp/TodosList.js:7]: also add propType

**Pull request merged**